tests/Backend/ExternalBuckets/DenyExternalBucketsInInputMappingTest.php

```diff
@@ -0,0 +1,183 @@
+            $this->manageClient->addProjectFeature($token['owner']['id'], self::PROJECT_FEATURE_DENY_EXTERNAL_DATASET_INPUT_MAPPING);
+        }
+
+        $workspaces = new Workspaces($this->_client);
+        $workspace0 = $workspaces->createWorkspace(['backend' => 'snowflake']);
+        $projectRole = $workspace0['connection']['database'];
+
+        $this->grantImportedPrivilegesToProjectRole($projectRole);
+
+        $description = $this->generateDescriptionForTestObject();
+        $testBucketName = $this->getTestBucketName($description);
+        $bucketId = self::STAGE_IN . '.' . $testBucketName;
+
+        $this->dropBucketIfExists($this->_client, $bucketId);
+
+        $this->_client->registerBucket(
+            $testBucketName,
+            explode('.', $this->getInboundSharedDatabaseName()),
+            self::STAGE_IN,
+            $description,
+            'snowflake',
+            $this->getName(),
+            true,
+        );
+
+        $registeredBucket = $this->_client->getBucket($bucketId);
+        $this->assertTrue($registeredBucket['isSnowflakeSharedDatabase']);
+
+        try {
+            $workspaces->loadWorkspaceData($workspace0['id'], [
+                'input' => [
+                    [
+                        'source' => $bucketId . '.NAMES_TABLE',
+                        'destination' => 'NAMES_TABLE_DESTINATION',
+                    ],
+                ],
+            ]);
+            $this->fail('Should fail');
+        } catch (ClientException $e) {
+            $this->assertStringContainsString(
+                'do not have enabled external schema support in Input Mapping',
+                $e->getMessage(),
+            );
+        }
+
+        $this->_client->dropBucket($bucketId);
+        $workspaces->deleteWorkspace($workspace0['id']);
+        $this->manageClient->removeProjectFeature($token['owner']['id'], self::PROJECT_FEATURE_DENY_EXTERNAL_DATASET_INPUT_MAPPING);
+    }
+
+    private function grantImportedPrivilegesToProjectRole(string $projectRole): void
+    {
+        $db = $this->ensureSnowflakeConnection();
+        $db->executeQuery('USE ROLE ACCOUNTADMIN');
+        $db->executeQuery(sprintf(
+            'GRANT IMPORTED PRIVILEGES ON DATABASE %s TO %s',
+            explode('.', $this->getInboundSharedDatabaseName())[0],
+            $projectRole,
+        ));
+    }
+
+    private function getInboundSharedDatabaseName(): string
+    {
+        $inboundDatabaseName = getenv('SNOWFLAKE_INBOUND_DATABASE_NAME');
+        assert($inboundDatabaseName !== false, 'SNOWFLAKE_INBOUND_DATABASE_NAME env var is not set');
+        $this->assertCount(
+            2,
+            explode('.', $inboundDatabaseName),
+            sprintf('SNOWFLAKE_INBOUND_DATABASE_NAME should have exactly 2 parts: <DATABASE_NAME>.<SCHEMA_NAME> gets %s', $inboundDatabaseName),
+        );
+        return $inboundDatabaseName;
+    }
```

Comment:
trochu blbé aby to byl c&p mezi těma třídama. Mohlo by to byt v trejtě.

Reply:
To je fakt. Zkusím to nějak rozumně sjednotit.